roles/allocations/tasks/main.yml

```
@@ -0,0 +1,47 @@
+---
+- name: Pull allocations service container images.
+  docker_image:
+    source: pull
+    name: "{{ item }}"
+    state: present
+    force_source: yes
+  loop:
+    - "{{ allocations_balance_service_docker_image }}"
+  tags:
+    - pull
+    
+- name: Create Docker network.
+  docker_network:
+    name: "{{ allocations_docker_network }}"
+    ipam_config:
+      - subnet: "{{ allocations_docker_network_subnet }}"
+    state: present
+  when: allocations_docker_network != "host"
+
+- name: Create init configuration.
+  template:
+    src: "allocations_balance_service.conf.j2"
+    mode: 0600
+    dest: "{{ allocations_config_path }}/service.conf"
+    owner: "{{ allocations_user }}"
+
+- name: Configure allocations services
+  include_role:
+    name: chameleon.docker_service
+  vars:
+    service_name: "{{ allocaitons_balance_service_name }}"
+    service_bind_address: "{{ ansible_facts[network_interface]['ipv4']['address'] }}"
+    docker_image: "{{ allocations_balance_service_docker_image }}"
+    docker_network: "{{ allocations_docker_network }}"
+    docker_mounts: 
+      - type: bind
+        src: "{{ allocations_config_path }}/service.conf"
+        dst: "/etc/allocations/balance-service.conf"
+    docker_ports: 
+      - "{{ allocations_balance_service_port }}:8080/tcp"
```

To avoid binding the port on all host interfaces, restrict to just the bind address:

```suggestion
      - "{{ api_interface_address }}:{{ allocations_balance_service_port }}:8080/tcp"
```